Misc/NEWS.d/next/Library/2022-05-25-15-57-39.gh-issue-90155.YMstB5.rst

```diff
@@ -0,0 +1 @@
+Fix broken :class:`asyncio.Semaphore` and strengthen FIFO guarantee.
```

Comment:
Can we qualify that it's broken when an acquire() operation is cancelled? And I'm not sure that we're actually strengthening the FIFO guarantee -- apart from the cancellation bug it was perfectly FIFO already, right?

Reply:
The `Lock` docs say `Lock` is FIFO. I didn't come up with a proof that the adapted `Semaphore` is FIFO (although I believe so). Yet the `Semephore` docs do not say it is FIFO, so we can't violate its docs.